piezo_web_app/PiezoWebApp/src/services/spark_job/spark_ui_service.py

```diff
@@ -0,0 +1,86 @@
+from urllib.parse import urlparse
+
+import kubernetes
+
+from PiezoWebApp.src.services.spark_job.i_spark_ui_service import ISparkUiService
+
+
+# ONLY CHANGE CONSTANTS IF YOU KNOW WHAT YOU ARE DOING
+UI_PORT = '4040'  # Port that the UI is exposed from the UI service (set by the spark operator)
+RELEASE_LABEL = 'piezo'  # Identifier label to allow for easy clean up
+PROXY_IMAGE = 'networkaispark/spark-ui-proxy:1.0.0'  # Docker image for spark ui proxy
+PROXY_RESOURCES = {'cpu': '500m'}  # Resources limits to allocate to each proxy pod
+PROBE_DELAY = 120   # Seconds before first check if proxy is running. Shuts down if not
+PROBE_TIMEOUT = 5   # Seconds after check if proxy is running times out
+HTTP_INGRESS_PORT = 80  # Default port for an ingress on http
+HTTPS_INGRESS_PORT = 443  # Default port for an ingress on https
+
+
+class SparkUiService(ISparkUiService):
+    def __init__(self, configuration):
+        self._k8s_url = configuration.k8s_url
+        self._proxy_port = HTTPS_INGRESS_PORT if urlparse(self._k8s_url).scheme == 'https' else HTTP_INGRESS_PORT
```

Comment:
Testing that the HTTP vs HTTPS distinction is made correctly will be much easier to test if it is part of `Configuration`. Look at `Configuration.is_s3_secure` and do the same for `Configuration.k8s_url`